packages/repository/test/acceptance/has-many-without-di.relation.acceptance.ts

```diff
@@ -0,0 +1,169 @@
+  before(givenCustomerRepository);
+  beforeEach(async () => {
+    existingCustomerId = (await givenPersistedCustomerInstance()).id;
+  });
+  afterEach(async () => {
+    await orderRepo.deleteAll();
+  });
+
+  it('can create an instance of the related model', async () => {
+    async function createCustomerOrders(
+      customerId: number,
+      orderData: Partial<Order>,
+    ): Promise<Order> {
+      return await customerRepo.orders({id: customerId}).create(orderData);
+    }
+    const order = await createCustomerOrders(existingCustomerId, {
+      description: 'order 1',
+    });
+    expect(order.toObject()).to.containDeep({
+      customerId: existingCustomerId,
+      description: 'order 1',
+    });
+
+    const persisted = await orderRepo.findById(order.id);
+    expect(persisted.toObject()).to.deepEqual(order.toObject());
+  });
+
+  it('can find instances of the related model', async () => {
+    async function createCustomerOrders(
+      customerId: number,
+      orderData: Partial<Order>,
+    ): Promise<Order> {
+      return await customerRepo.orders({id: customerId}).create(orderData);
+    }
+    async function findCustomerOrders(customerId: number) {
+      return await customerRepo.orders({id: customerId}).find();
+    }
+
+    const order = await createCustomerOrders(existingCustomerId, {
+      description: 'order 1',
+    });
+    const notMyOrder = await createCustomerOrders(existingCustomerId + 1, {
+      description: 'order 2',
+    });
+    const orders = await findCustomerOrders(existingCustomerId);
+
+    expect(orders).to.containEql(order);
+    expect(orders).to.not.containEql(notMyOrder);
+
+    const persisted = await orderRepo.find({
+      where: {customerId: existingCustomerId},
+    });
+    expect(persisted).to.deepEqual(orders);
+  });
+
+  //--- HELPERS ---//
+
+  @model()
+  class Order extends Entity {
+    @property({
+      type: 'number',
+      id: true,
+    })
+    id: number;
+
+    @property({
+      type: 'string',
+      required: true,
+    })
+    description: string;
+
+    @property({
+      type: 'number',
+      required: true,
+    })
+    customerId: number;
+  }
+
+  @model()
+  class Customer extends Entity {
+    @property({
+      type: 'number',
+      id: true,
+    })
+    id: number;
+
+    @property({
+      type: 'string',
+    })
+    name: string;
+
+    @hasMany({keyTo: 'customerId'})
+    @property.array(Order)
+    orders: Order[];
+  }
+
+  class OrderRepository extends DefaultCrudRepository<
+    Order,
+    typeof Order.prototype.id
+  > {
+    constructor(db: juggler.DataSource) {
+      super(Order, db);
+    }
+  }
+
+  class CustomerRepository extends DefaultCrudRepository<
+    Customer,
+    typeof Customer.prototype.id
+  > {
+    public orders: HasManyRepositoryFactory<Customer, Order>;
+
+    constructor(
+      protected db: juggler.DataSource,
+      orderRepository: DefaultCrudRepository<Order, typeof Order.prototype.id>,
```

Comment:
`DefaultCrudRepository` - is that an interface or a class? The API should be accepting interfaces, it's up to the caller to provide an implementation (class) that matches the required interface.